Objects/longobject.c

```diff
@@ -3028,33 +3028,24 @@ PyLong_AsDouble(PyObject *v)
 
 /* Methods */
 
-static int
+static Py_ssize_t
 long_compare(PyLongObject *a, PyLongObject *b)
 {
-    Py_ssize_t sign;
-
-    if (Py_SIZE(a) != Py_SIZE(b)) {
-        sign = Py_SIZE(a) - Py_SIZE(b);
-    }
-    else {
+    Py_ssize_t sign = Py_SIZE(a) - Py_SIZE(b);
+    if (!sign) {
         Py_ssize_t i = Py_ABS(Py_SIZE(a));
-        while (--i >= 0 && a->ob_digit[i] == b->ob_digit[i])
+        sdigit diff = 0;
+        while (--i >= 0 && !(diff = (sdigit) a->ob_digit[i] - (sdigit) b->ob_digit[i]))
```

Comment:
Too long line.

I would rewrite this as:
```c
while (--i >= 0) {
    sdigit diff = ...;
    if (diff) {
        sign = ...;
        break;
    }
}
```

Reply:
Ok, this does not affect the instructions.